packages/keyring-controller/src/KeyringController.ts

```diff
@@ -2266,22 +2356,25 @@ export class KeyringController extends BaseController<
   async #removeEmptyKeyrings(): Promise<void> {
     this.#assertControllerMutexIsLocked();
     const validKeyrings: EthKeyring<Json>[] = [];
+    const validKeyringMetadata: KeyringMetadata[] = [];
 
     // Since getAccounts returns a Promise
     // We need to wait to hear back form each keyring
     // in order to decide which ones are now valid (accounts.length > 0)
 
     await Promise.all(
-      this.#keyrings.map(async (keyring: EthKeyring<Json>) => {
+      this.#keyrings.map(async (keyring: EthKeyring<Json>, index: number) => {
```

Comment:
The way we are using this `.map` is not ideal. I'm not sure what this `await Promise.all` is supposed to do since the map doesn't return anything - I believe that each map callback function here is executed before `Promise.all` receives any promise handle, but even if this code worked as intended it would be very dangerous to destroy keyrings in a random order like this

Though this code was here already, so we can also fix this in another PR